src/App.tsx

```diff
@@ -1,19 +1,152 @@
 import React from 'react';
-import './App.css';
+import './App.scss';
 
 import users from './api/users';
+import todos from './api/todos';
+
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+const preparedTodos = todos.map((todo) => {
+  return {
+    ...todo,
+    user: users.find(user => user.id === todo.userId) || null,
+  };
+});
+
+type State = {
+  todosList: Todo[],
+  taskText: string,
+  selectedUser: string,
+  userNotSelectedMessage: boolean,
+  titleIsEmpty: boolean,
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
 };
 
+class App extends React.Component<{}, State> {
+  state = {
+    todosList: [...preparedTodos],
+    taskText: '',
+    selectedUser: '',
+    userNotSelectedMessage: false,
+    titleIsEmpty: false,
+  };
+
+  clearForm = () => {
+    this.setState({ taskText: '', selectedUser: '' });
+  };
+
+  validateForm = () => {
+    if (this.state.selectedUser !== '') {
+      this.setState({ userNotSelectedMessage: false });
+    } else {
+      this.setState({ userNotSelectedMessage: true });
+    }
+
+    if (this.state.taskText !== '') {
+      this.setState({ titleIsEmpty: false });
+    } else {
+      this.setState({ titleIsEmpty: true });
+    }
+  };
+
+  handleTitleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      taskText: event.target.value,
+      titleIsEmpty: false,
+    });
+  };
+
+  handleSelectField = (event: { target: { value: string; }; }) => {
+    const { value } = event.target;
+
+    this.setState({
+      selectedUser: value,
+      userNotSelectedMessage: false,
+    });
+  };
+
+  addTodo = () => {
+    const { selectedUser } = this.state;
+
+    const currentUser: User | null
+      = users.find((user) => user.name === selectedUser) || null;
+
+    if (!currentUser) {
+      return;
+    }
+
+    this.setState((state) => ({
+      todosList: [...state.todosList, {
+        userId: 1,
+        id: state.todosList.length + 1,
+        title: state.taskText,
+        completed: false,
+        user: currentUser,
+      }],
+    }));
+
+    this.clearForm();
+  };
+
+  render() {
+    const {
+      todosList, taskText, selectedUser, userNotSelectedMessage, titleIsEmpty,
+    } = this.state;
+
+    return (
+      <div className="App">
+        <h1 className="App__title">Add todo form</h1>
+        <form onSubmit={event => event.preventDefault()}>
+          <div className="App__taskTitle">
+            <input
+              type="text"
+              size={22}
+              value={taskText}
+              placeholder="Please enter the title"
+              onChange={(event) => this.handleTitleInput(event)}
+            />
+            <p className={titleIsEmpty ? 'App__warning-text' : 'App__warning-text--none'}>
```

Comment:
Have you ever used the `classNames` libary? Great things for conditional classes. Take a look)

Reply:
I used it before, but this time I had issues with it on this current task so decided not to include it)